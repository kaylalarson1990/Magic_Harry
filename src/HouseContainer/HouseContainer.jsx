import React, { Component } from "react";
import { connect } from "react-redux";
import "./Container.scss";
import Houses from "../Houses/Houses.jsx";
import PropTypes from "prop-types";

export class HouseContainer extends Component {
  constructor() {
    super();
    this.state = {
      error: "",
      filtered: ""
    };
  }

  handleChange = (e) => {
    this.setState({
      filtered: e.target.value
    })
  }

  render() {
    let filteredHouses = this.props.houses.filter(house => {
      return house.name.toLowerCase().indexOf(this.state.filtered) !== -1;
    });
    const displayHouses = filteredHouses.map(house => (
      <Houses
        key={house.id}
        name={house.name}
        mascot={house.mascot}
        headOfHouse={house.headOfHouse}
        school={house.school}
        houseGhost={house.houseGhost}
        founder={house.founder}
      />
    ));
    return (
      <>
        <article>
          <h2 className="title">Harry Potter Houses</h2>
          <input className="search__input" type="text" value={this.state.filtered} onChange={this.handleChange} placeholder="Search" />
        </article>
        <section className="houses">{displayHouses}</section>
      </>
    );
  }
}

export const mapStateToProps = state => ({
  houses: state.houses
});


HouseContainer.propTypes = {
  houses: PropTypes.array,
  setHouses: PropTypes.func
};

export default connect(
  mapStateToProps
)(HouseContainer);
