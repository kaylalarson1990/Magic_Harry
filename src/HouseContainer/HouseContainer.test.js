import { HouseContainer, mapStateToProps } from "./HouseContainer.jsx";
import { shallow } from "enzyme";
import React from "react";

describe(HouseContainer, () => {
  let wrapper, mockData, mockSubmit;
  beforeEach(() => {
    mockData = [
      {
        id: "1",
        name: "Kayla",
        mascot: "Bengal",
        headOfHouse: "Klio Pold",
        school: "Turing",
        houseGhost: "Dobby",
        founder: "Sergio"
      }
    ];

    mockSubmit = jest.fn();

    wrapper = shallow(<HouseContainer houses={mockData} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setState on filtered", () => {
    const mockEvent = {
      target: {
        value: "Kayla"
      }
    };

    const expected = {
      error: "",
      filtered: "Kayla"
    };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a house array", () => {
    const mockState = {
      houses: "Apartment complex",
      type: "SET_HOUSES"
    };
    const expected = {
      houses: "Apartment complex"
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
