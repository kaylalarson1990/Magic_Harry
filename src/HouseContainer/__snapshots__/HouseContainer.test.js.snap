// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseContainer should match snapshot 1`] = `
<Fragment>
  <article>
    <h2
      className="title"
    >
      Harry Potter Houses
    </h2>
    <input
      className="search__input"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </article>
  <section
    className="houses"
  >
    <Houses
      founder="Sergio"
      headOfHouse="Klio Pold"
      houseGhost="Dobby"
      key="1"
      mascot="Bengal"
      name="Kayla"
      school="Turing"
    />
  </section>
</Fragment>
`;
